core/primitives/src/validator_signer.rs

```diff
@@ -14,53 +15,114 @@ use crate::stateless_validation::{
 use crate::telemetry::TelemetryInfo;
 use crate::types::{AccountId, BlockHeight, EpochId};
 
+#[derive(Clone, Debug, PartialEq)]
+pub enum ValidatorSigner {
+    EmptyValidatorSigner(EmptyValidatorSigner),
```

Comment:
simplify the enum names as Empty(...) and InMemory(...)? the struct name and arguments already talk about ValidatorSigner.

Reply:
You mean `ValidatorSigner::Empty(EmptyValidatorSigner)`, etc. ?

Reply:
yes

Reply:
Great idea! Thanks